tianshou/highlevel/module/actor.py

```diff
@@ -0,0 +1,261 @@
+from abc import ABC, abstractmethod
+from collections.abc import Sequence
+from dataclasses import dataclass
+from enum import Enum
+from typing import Protocol
+
+import torch
+from sensai.util.string import ToStringMixin
+from torch import nn
+
+from tianshou.highlevel.env import Environments, EnvType
+from tianshou.highlevel.module.core import (
+    ModuleFactory,
+    TDevice,
+    init_linear_orthogonal,
+)
+from tianshou.highlevel.module.intermediate import (
+    IntermediateModule,
+    IntermediateModuleFactory,
+)
+from tianshou.highlevel.module.module_opt import ModuleOpt
+from tianshou.highlevel.optim import OptimizerFactory
+from tianshou.utils.net import continuous, discrete
+from tianshou.utils.net.common import BaseActor, ModuleType, Net
+
+
+class ContinuousActorType(Enum):
+    GAUSSIAN = "gaussian"
+    DETERMINISTIC = "deterministic"
+    UNSUPPORTED = "unsupported"
+
+
+@dataclass
+class ActorFuture:
```

Comment:
I think it's a fitting name, because it's a realization of the same fundamental concept of a "future", i.e. an object that can provide a value in the future. In this case, it's just a simple container, because there is no asynchronicity involved.
Note that this is an internal class which library users will not be interacting with.

Do you have another suggestion?